### PR TITLE
Update devcontainer

### PR DESCRIPTION
**Pull request by zachlankton**
## Overview

This PR is to bring the `devcontainer` in this repo up to date and improve its reliability.  The current config uses outdated dependencies and does not work out of the box.

Here are the updates:

- Use volume mount instead of bind to improve performance and help solve #4145 
- Updated `README` and `README-DEV` to include quick buttons/links to spin up devcontainer
- Automatically run `./configure && make` during container creation

## Testing performed to validate dev container...

Click the button below to automatically spin up a devcontainer on your local machine.

(this demo button uses [zachlankton/couchdb-test](https://github.com/zachlankton/couchdb-test) repo to show how the button will work once this PR is merged into main)

[Open in Remote - Containers](https://vscode.dev/redirect?url=vscode://ms-vscode-remote.remote-containers/cloneInVolume?url=https://github.com/zachlankton/couchdb-test)

Once the container is finished setting up (will take some time due to automatically running `./configure && make` run the following commands to verify everything works:

```bash
./dev/run
# or ./dev/run --admin=admin:admin
# or ./dev/run --with-admin-party-please

# login to http://localhost:15984/_utils/
# make a db and some docs
# run the "Verify Install"

# CTRL-C when you are satisfied
# then run
make check

# all test should pass!
```

## Checklist

- [X] Code is written and works correctly
- [n/a] Changes are covered by tests
- [n/a] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [n/a] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,24 +1,13 @@
-ARG ELIXIR_VERSION
-FROM elixir:${ELIXIR_VERSION}
+ARG IMG
+FROM ${IMG}
```
Comment (big-r81) [inline .devcontainer/Dockerfile:2]: Maybe change the variable name to something more descriptive, like 'COUCHDB_IMAGE' or whatever you like.

File: README-DEV.rst [hunk 2]
```diff
@@ -64,6 +64,21 @@ associated ``devcontainer.json`` file to quickly provision a
...
+
+If you already have VS Code and Docker installed, you can click the badge above or 
+`here <https://vscode.dev/redirect?url=vscode://ms-vscode-remote.remote-containers/cloneInVolume?url=https://github.com/apache/couchdb>`_ 
+to get started. Clicking these links will cause VS Code to automatically install the 
+Remote - Containers extension if needed, clone the source code into a container volume, 
+and spin up a dev container for use.
+
+This ``devcontainer`` will automatically run ``./configure && make`` the first time it is created.  
+While this may take some extra time to spin up, this tradeoff means you will be able to 
+run things like ``./dev/run`` and ``make check`` straight away.  Subsequent startups should be quick.
```
Comment (big-r81) [inline README-DEV.rst:79]: I think we need to give `./dev/run` and admin user with password or the switch `--with-admin-party-please`. Do you tried it with only `./dev/run`?

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (zachlankton) [reply .devcontainer/Dockerfile:2]: Will do!

File: README-DEV.rst (hunk 2 above)
Comment (zachlankton) [reply README-DEV.rst:79]: It works with all of the above, I've tried them all.  Would you like this to be updated to show all three examples? Or just one of them?

**Review by nickva — APPROVED** [review]
Thank you for your contribution @zachlankton

+1

Installed vscode on ubuntu 20.04 and managed to quickly get to a point where I could run `make eunit` tests.

**Pull request merged**